internal/controller/snapshot/snapshot_controller.go

```diff
@@ -112,6 +113,15 @@ func (r *Reconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Resu
 		return helpers.HandleLoaderError(logger, err, "Application", "Snapshot")
 	}
 
+	if !controllerutil.HasControllerReference(snapshot) {
+		snapshot, err = gitops.SetOwnerReference(ctx, r.Client, snapshot, application)
+		if err != nil {
+			logger.Error(err, fmt.Sprintf("Failed to set owner reference for snapshot %s/%s", snapshot.Namespace, snapshot.Name))
+			return ctrl.Result{}, err
+		}
+		logger.Info(fmt.Sprintf("Application %s has been set as a Controller OwnerReference on Snapshot %s", application.Name, snapshot.Name))
```

Comment:
Maybe we can change this to a logger.LogAuditEvent call since we're updating the object on the cluster here?
Something like (untested):
```
logger.LogAuditEvent(fmt.Sprintf("Failed to set owner reference for snapshot %s/%s", snapshot.Namespace, snapshot.Name),
	snapshot, helpers.LogActionUpdate)
```

Reply:
Changed to log as audit event, it looks like this now:
```
{"level":"info","ts":"2024-12-06T11:50:19Z","logger":"controllers.snapshot","caller":"snapshot/snapshot_controller.go:122","msg":"Application application-sample has been set as a Controller OwnerReference on Snapshot snapshot-normal","snapshot":{"name":"snapshot-normal","namespace":"default"},"namespace":"default","name":"snapshot-normal","controllerKind":"Snapshot","audit":"true","action":"UPDATE"}
```

Please lmk if this is what you were suggesting.

Reply:
Yup, that's it, thanks!